### PR TITLE
feat(donation-grid): Donation form grid display option #1414

### LINKED ISSUE #1414: feat(grid): display multiple forms in a grid
## Issue Overview
We should have a `[donation_form_grid]` shortcode that display donation forms in a nice layout:
grid

### Shortcode Attributes
- `number`: Default = `3`
- `goal` (if applicable): Default = `true`
- `excerpt`: Default = `true`
- `featured_image` : Default = `true`

### Acceptance Criteria
- Grid should fill full width of container
- Grid should be responsive
- Grid should have minimal conflict with themes

## Tasks
- [x] Design Grid
- [x] Develop Shortcode
- [ ] Implement CSS Grid
- [ ] Test with Popular Themes (Avada, Divi, X, etc)
- [ ] Documentation

### PR DESCRIPTION
**Pull request by Sidsector9**
Closes #1414 

## Description

1. Added `[give_donation_grid]` shortcode
2. Added following attributes to the shortcode
    1. `columns` (1|2|3|4): Sets the number of grid items per row
    2. `show_goal` (true|false): Display progress bar if goal is set
    3. `show_excerpt` (true|false): Display excerpt if it is enabled from Give Settings and shortcode.
    4. `show_featured_image` (true|false): Display featured image if it is enabled from Give Settings and shortcode.
    5. `display_type` (redirect|modal): Whether to redirect to a new page _or_ to open a modal box after clicking **Donate Now**

_[Design is kept as minimal until further discussion]_

## How Has This Been Tested?
This is WIP. Only manual testing done as of 7th Feb 2018

### Desktop
desktop view

### Tablet
tablet view

### Video demo
https://www.useloom.com/share/edd2bdcf0cef486ea4676fbe1ca180e2

### CODE REVIEW

File: templates/shortcode-donation-grid.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		// The 'Donate Now' button.
+		if ( 'redirect' === $atts['display_type'] && 'redirect' === $atts['display_type'] ) {
+
+			// 'Donate Now' button if the 'display_type' attribute is set to 'redirect'
+			printf( '<a class="grid-donate-now" href="%1$s">%2$s</a>', get_the_permalink(), apply_filters( 'donation_grid_donate_now', esc_html__( 'Donate Now', 'grid' ) ) );
+
+		} elseif( 'modal' === $atts['display_type'] ) {
+
+			// 'Donate Now' button if the 'display_type' attribute is set to 'modal'
+			printf( '<a class="grid-donate-now grid-donate-now-modal-button" data-effect="mfp-zoom-out" href="#popup-form-%1$s">%2$s</a>', get_the_ID(), apply_filters( 'donation_grid_donate_now', esc_html__( 'Donate Now', 'grid' ) ) );
```
Comment (DevinWalker) [inline templates/shortcode-donation-grid.php]: @Sidsector9 can you make the effect a slide down/up to match how to works with the "Modal" Display Mode? You'll see it slides in and out sort of like bootstrap modals.

**Comment by DevinWalker** [comment]
@Sidsector9 can we tighten up the spacing, add a bit more padding to the content, make the button full width, and basically make it look a bit prettier? Also, have you cross-theme tested this much? Thanks for the video that was helpful but I'd like to see how it looks across a few themes.

68747470733a2f2f736e61672e67792f3569514437462e6a7067

**Comment by Sidsector9** [comment]
@DevinWalker I changed the look, I have added a little animation so that the grid items look lively, let me know you thoughts on this - https://www.useloom.com/share/127f5e3d20924d7882dcdb75bf798baf

File: templates/shortcode-donation-grid.php (hunk 1 above)
Comment (Sidsector9) [reply templates/shortcode-donation-grid.php]: @DevinWalker Updated the animation

**Comment by Sidsector9** [comment]
@DevinWalker The changes are done on this one as you asked. Can you review it?

**Pull request merged**